pack/map.h

```diff
@@ -339,6 +357,20 @@ const Attribute& Map<T>::get(const std::string& key) const
     throw std::out_of_range("Key " + key + " was not found");
 }
 
+template <typename T>
+void Map<T>::set(const std::string& key, T& val)
+{
+    auto found = std::find_if(m_value.begin(), m_value.end(), [&](const auto& pair) {
+        return pair.first == key;
+    });
+
+    if (found != m_value.end()) {
+        found->second = val;
+    } else {
+        throw std::out_of_range("Key " + key + " was not found");
```

Comment:
I don't get it. If now such key in map, throw exception? Why? We need to insert it to the map, right?